node/src/components/fetcher.rs

```diff
@@ -293,13 +293,21 @@ impl ItemFetcher<GlobalStorageTrie> for Fetcher<GlobalStorageTrie> {
         id: Digest,
         peer: NodeId,
     ) -> Effects<Event<GlobalStorageTrie>> {
-        effect_builder
-            .read_trie(id)
-            .event(move |maybe_trie| Event::GetFromStorageResult {
+        async move {
+            let maybe_trie = match effect_builder.read_trie(id).await {
+                Ok(maybe_trie) => maybe_trie,
+                Err(error) => {
+                    error!(?error, "read_trie_request");
+                    None
+                }
```

Comment:
Why not return the result to the caller and let it decide what to do about the error?

Reply:
The type changed from `Option<_>`, to `Result<Option<_>>` and I think this was done here to minimize changes to the caller.